catroid/src/org/catrobat/catroid/physics/PhysicsObject.java

```diff
@@ -333,6 +328,19 @@ protected void setCollisionBits(short categoryBits, short maskBits) {
 			filter.maskBits = maskBits;
 			fixture.setFilterData(filter);
 		}
+
+		if (updateState) {
+			updateNonCollidingState();
+		}
+	}
+
+	private void updateNonCollidingState() {
+		if (body.getUserData() != null && body.getUserData() instanceof Sprite) {
+			Object look = ((Sprite) body.getUserData()).look;
+			if (look != null && look instanceof PhysicsLook) {
+				((PhysicsLook) look).setNonColliding(collisionMaskRecord == PhysicsWorld.MASK_NOCOLLISION);
+			}
```

Comment:
Maybe its possible to save the PhysicsObjectStateHandler instead of the Sprite into the bodies userdata. But I am afraid that the Sprite is necessary anywhere else.
On the other side, the body knows the sprite anyway, so why not use it?